service/src/test/java/uk/gov/hmcts/reform/fpl/service/removeorder/RemoveOrderServiceTest.java

```diff
@@ -115,46 +127,68 @@ void shouldMakeDynamicListOfMixedOrderTypes() {
 
     @Test
     void shouldUseGeneratedOrderRemovalActionWhenRemovingGeneratedOrder() {
-        when(orderRemovalActions.getAction(REMOVED_UUID, removableOrder)).thenReturn(generatedOrderRemovalAction);
+        when(orderRemovalActions.getAction(removableOrder)).thenReturn(generatedOrderRemovalAction);
 
         underTest.removeOrderFromCase(caseData, data, REMOVED_UUID, removableOrder);
 
         verify(generatedOrderRemovalAction).remove(caseData, data, REMOVED_UUID, removableOrder);
         verifyNoMoreInteractions(generatedOrderRemovalAction);
-        verifyNoInteractions(cmoOrderRemovalAction);
+        verifyNoInteractions(cmoRemovalAction, sdoRemovalAction);
     }
 
     @Test
     void shouldUseCMORemovalActionWhenRemovingCMO() {
-        when(orderRemovalActions.getAction(REMOVED_UUID, removableOrder)).thenReturn(cmoOrderRemovalAction);
+        when(orderRemovalActions.getAction(removableOrder)).thenReturn(cmoRemovalAction);
 
         underTest.removeOrderFromCase(caseData, data, REMOVED_UUID, removableOrder);
 
-        verify(cmoOrderRemovalAction).remove(caseData, data, REMOVED_UUID, removableOrder);
-        verifyNoMoreInteractions(cmoOrderRemovalAction);
-        verifyNoInteractions(generatedOrderRemovalAction);
+        verify(cmoRemovalAction).remove(caseData, data, REMOVED_UUID, removableOrder);
+        verifyNoMoreInteractions(cmoRemovalAction);
+        verifyNoInteractions(generatedOrderRemovalAction, sdoRemovalAction);
+    }
+
+    @Test
+    void shouldUseSDORemovalActionWhenRemovingSDO() {
+        when(orderRemovalActions.getAction(removableOrder)).thenReturn(sdoRemovalAction);
+
+        underTest.removeOrderFromCase(caseData, data, REMOVED_UUID, removableOrder);
+
+        verify(sdoRemovalAction).remove(caseData, data, REMOVED_UUID, removableOrder);
+        verifyNoMoreInteractions(sdoRemovalAction);
+        verifyNoInteractions(generatedOrderRemovalAction, cmoRemovalAction);
     }
 
     @Test
     void shouldUseGeneratedOrderPopulateCaseFieldActionWhenPopulatingCaseFieldsForGeneratedOrder() {
-        when(orderRemovalActions.getAction(REMOVED_UUID, removableOrder)).thenReturn(generatedOrderRemovalAction);
+        when(orderRemovalActions.getAction(removableOrder)).thenReturn(generatedOrderRemovalAction);
```

Comment:
I don't think you need to test the specific actions. Those should be tested separately. you can assume that  orderRemovalActions.getAction(removableOrder) should return you the right action(a mock implementing the OrderRemovalAction)